obspy/io/segy/core.py

```diff
@@ -200,10 +205,15 @@ def _read_segy(filename, headonly=False, byteorder=None,
 
     # Convert traces to ObsPy Trace objects.
     for tr in segy_object.traces:
-        stream.append(tr.to_obspy_trace(
-            headonly=headonly,
-            unpack_trace_headers=unpack_trace_headers))
-
+        try:
+            stream.append(tr.to_obspy_trace(
+                headonly=headonly,
+                unpack_trace_headers=unpack_trace_headers))
+        except ValueError:
+            #skip the remaining corrupt traces
+            if(not skip_corrupt_traces):raise ValueError
+            break
```

Comment:
this should probably be

```
except ValueError as e:
    if not skip_corrupt_traces:
        raise e
```

I don't understand the `break`, isn't the intent of the PR that reading continues after a corrupt trace?

Reply:
Good catch, that is a mistake left over from earlier testing (that break causes it to stop so you get a stream object with all of the good traces up until a corrupt trace)

Reply:
I ended up leaving this as is. I explain why below